### DISCUSSION TITLE
How to add/ enable irq in ports/minimal code

### DISCUSSION THREAD

**Discussion by synapticonashaikh** [micropython/micropython#10295] [STM32 / Pyboard]
Hello, 
I'm working on a project where I'm building a minimal version of the micropython stack so that it fits on my board (STM32F446RC, 256kb flash). I would like to know what files I need to add to use Timer and IRQ (callbacks). I tried copying many files from the ports/stm32, but unfortunately, I can't make it work. Does anyone know what files I should copy and what changes I need to make in my makefile to get it working?

**Comment by jimmo** [comment]
You will find it much easier to remove stuff from the stm32 port rather than adding stuff to minimal.

There are some existing boards in the stm32 port that have less than 256kiB flash (e.g. NUCLEO_F091RC or B_L072Z_LRWAN1).

**Reply by synapticonashaikh** [reply]
I tried with an example board of STM32F446RE(512Kb)! But as the definition (to enable/disable) spreads among so many files, it is hard to track where I can disable the extra modules, which I don't need. But yes, I'll try these smaller flash-size boards you mentioned to see if they fulfil my requirements.
